sympy/core/expr.py

```diff
@@ -3787,7 +3787,12 @@ def unchanged(func, *args):
     False
     """
     f = func(*args)
-    return f.func == func and f.args == tuple([sympify(a) for a in args])
+    types = [type(arg) for arg in f.args]
+    try:
+        cargs = tuple([cast(a) for a, cast in zip(args, types)])
+    except:
```

Comment:
Bare `except` catches everything, including system exit things like `KeyboardInterrupt`. It should be avoided.

Reply:
Ah! I didn't know that, thanks for telling this. I will update the PR.